platform/engineering/request-for-comment/2020-11-4-storybook-hosting.md

```diff
@@ -0,0 +1,70 @@
+# RFC: Host Storybook on design.va.gov
+
+- Date: 2020-11-4
+- Related Issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/15068
+
+## Background
+We have component library documentation in two places:
+- https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/visual-design
+- https://design.va.gov/components
+
+We're going to use Storybook to consolidate the documentation into one place:
+design.va.gov.
+
+design.va.gov is a Jekyll site. The documentation in it is manually updated. The
+repo can be found at
+[`vets-design-system-documentation`](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/).
+It's deployed daily by <this Jenkins job> to the S3 bucket from which the static
+site is served.
+
+## Motivation
+We want to serve the component library documentation from Storybook on
+design.va.gov to ensure the documentation is in a single place that's
+automatically updated when updates are made to the library. We chose Storybook
+as the tool to keep the documentation closely tied to the code itself.
+
+## Scope
+Outside the scope of this document is the code location of the component library
+and Storybook. While the location of the code will impact where the CI
+configuration is kept, the solution outlined here must be possible whether the
+code lives in
+[`veteran-facing-services-tools`](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools),
+[`vets-website`](https://github.com/department-of-veterans-affairs/vets-website),
+or its own repo, as we may move it later.
+
+For now, however, we're planning on moving the component library to its own repo.
+
+## Design
+- Storybook will be deployed to the S3 bucket from which design.va.gov is served
+  - **Question:** How will we be able to ensure deploying the Jekyll site won't
+    override the Storybook directory? What [`aws
+    s3`](https://docs.aws.amazon.com/cli/latest/reference/s3/) command do we
+    use?
+- Deployment will happen at the end of a successful `master` branch build if
+  there have been changes to the library
+  - This means continual deployment, which is a slight deviation of the pattern
+    we use for all other deploys
```

Comment:
All good. @dginther and I have ongoing conversations about incrementally moving toward continuous-deployment.